src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleTypeSymbol.vb

```diff
@@ -0,0 +1,987 @@
+                    Dim fieldRemainder As Integer ' one - based
+                    Dim fieldChainLength = NumberOfValueTuples(i + 1, fieldRemainder)
+                    Dim container As NamedTypeSymbol = GetNestedTupleUnderlyingType(_underlyingType, fieldChainLength - 1).OriginalDefinition
+
+                    Dim DiagnosticInfo = If(container.IsErrorType(),
+                                                          Nothing,
+                                                          ErrorFactory.ErrorInfo(ERRID.ERR_MissingRuntimeHelper,
+                                                                               container.Name & "." & TupleMemberName(fieldRemainder)))
+
+                    Dim defaultName As String = TupleMemberName(i + 1)
+                    ' Add a field with default name if the given name Is different
+                    If Name <> defaultName Then
+                        members.Add(New TupleErrorFieldSymbol(Me, defaultName, -members.Count - 1, Nothing, _elementTypes(i), DiagnosticInfo))
+                    End If
+
+                    members.Add(New TupleErrorFieldSymbol(Me, Name, i,
+                                                      If(_elementLocations.IsDefault, Nothing, _elementLocations(i)),
+                                                      _elementTypes(i),
+                                                      DiagnosticInfo))
+                End If
+            Next
+
+            Return members.ToImmutableAndFree()
+        End Function
+
+        Private Shared Sub CollectTargetTupleFields(underlying As NamedTypeSymbol, fieldsForElements As ArrayBuilder(Of FieldSymbol))
+            underlying = underlying.OriginalDefinition
+            Dim num As Integer = Math.Min(underlying.Arity, TupleTypeSymbol.RestPosition - 1)
+            For i As Integer = 0 To num - 1
+                Dim tupleTypeMember As WellKnownMember = TupleTypeSymbol.GetTupleTypeMember(underlying.Arity, i + 1)
+                fieldsForElements.Add(CType(TupleTypeSymbol.GetWellKnownMemberInType(underlying, tupleTypeMember), FieldSymbol))
+            Next
+        End Sub
+
+        Private Function ComputeDefinitionToMemberMap() As SmallDictionary(Of Symbol, Symbol)
+            Dim smallDictionary As SmallDictionary(Of Symbol, Symbol) = New SmallDictionary(Of Symbol, Symbol)(ReferenceEqualityComparer.Instance)
+            Dim originalDefinition As NamedTypeSymbol = Me._underlyingType.OriginalDefinition
+            Dim members As ImmutableArray(Of Symbol) = Me.GetMembers()
+            Dim i As Integer = members.Length - 1
+            While i >= 0
+                Dim symbol As Symbol = members(i)
+                Dim kind As SymbolKind = symbol.Kind
+                Select Case kind
+                    Case SymbolKind.[Event]
+                        Dim tupleUnderlyingEvent As EventSymbol = (DirectCast(symbol, EventSymbol)).TupleUnderlyingEvent
```

Comment:
done

---
In reply to: [71570753](https://github.com/dotnet/roslyn/pull/12596#discussion_r71570753) [](ancestors = 71570753)